src/ansys/fluent/core/session_meshing.py

```diff
@@ -66,3 +71,130 @@ def PMFileManagement(self):
+
+    def read_case(
+        self,
+        file_name: str,
+    ):
+        """Reads a case file.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+        """
+        if not self._server_file_manager:
+            self._server_file_manager = _ServerFileManager(self._fluent_connection)
+        return self._server_file_manager.read_case(
+            file_name,
+        )
+
+    def write_case(
+        self,
+        file_name: str,
+    ):
+        """Writes a case file.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+        """
+        if not self._server_file_manager:
+            self._server_file_manager = _ServerFileManager(self._fluent_connection)
+        return self._server_file_manager.write_case(file_name)
+
+
+class _ServerFileManager(Meshing):
+    """Supports file upload and download for every existing file read,
+    write methods respectively in the cloud particularly in Ansys lab.
+    Here we are supporting upload and download methods in existing session
+    methods. These would be no-ops if PyPIM is not configured or not authorized
+    with the appropriate service. This will be used for internal purpose only.
+
+    Methods
+    -------
+    read_case(
+        file_name
+        )
+        Read a case file.
+
+    write_case(
+        file_name
+        )
+        Write a case file.
+    """
+
+    def __init__(self, fluent_connection):
+        super(_ServerFileManager, self).__init__(fluent_connection)
+        self.pim_instance = fluent_connection._remote_instance
+        self.file_service = None
+        try:
+            upload_server = self.pim_instance.services["http-simple-upload-server"]
+        except (AttributeError, KeyError):
+            pass
+        else:
+            from simple_upload_server.client import Client
+
+            self.file_service = Client(
+                token="token", url=upload_server.uri, headers=upload_server.headers
+            )
+
+    def _wait_for_file(self, file_name, file_service):
+        start_time = time.time()
+        max_wait_time = 100
+        while (time.time() - start_time) < max_wait_time:
+            if file_service.file_exist(os.path.basename(file_name)):
+                break
+            max_wait_time -= 1
+            time.sleep(3)
+        else:
+            raise FileNotFoundError(f"{file_name} does not exist.")
+
+    def read_case(
+        self,
+        file_name: str,
+    ):
+        """Reads  a case file.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+
+        Raises
+        ------
+        FileNotFoundError
+            If a case file does not exist.
+        """
+        if pypim.is_configured():
+            if os.path.isfile(file_name):
+                if not self.file_service.file_exist(os.path.basename(file_name)):
+                    self.upload(os.path.basename(file_name))
```

Comment:
I'm curious about how this line works. If I have some `file_name` that specifies a file that can be at any path on the local machine, and I just strip it down to its base name and ask for the file of that name to be uploaded, what's actually happening there?

Reply:
It is required due to following logic used in `client.py`.

![image](https://github.com/ansys/pyfluent/assets/106588300/b2713085-9c4a-43f8-8a8e-eff0b5b92427)

If we pass `file_name` as complete file path to `self.file_service.file_exist()` , then it shows that file does not exist on the server.

Therefore we are reducing it to base name.